app/app/api/api_v1/endpoints/users.py

```diff
@@ -146,17 +145,26 @@ def update_user(
     """
 
     # Fetch existing User object from db
-    user = crud.user.get(db, id=user_id)
-    if not user:
-        raise HTTPException(
-            status_code=404,
-            detail="The user with this id does not exist in the system",
-        )
 
-    # Save updated object based on given UserUpdate object in db
-    user = crud.user.update(db, db_obj=user, obj_in=user_in)
+    if user := crud.user.get(db, id=user_id):
+        if user_in.type:
+            if user.type not in ('admin', 'professor'):
+                raise HTTPException(
+                    status_code=400,
+                    detail=f"A {user.type}'s role cannot be changed!",
+                )
+            if user_in.type not in ('admin', 'professor'):
```

Comment:
when an admin is promoted, are we going to change their type as well or just keep the type as professor and set `is_admin` to true? Changing the user type doesn't seem to be a great idea tbh, considering that they'll still have to have access to professor module as well.

Reply:
You mean when a professor is promoted?
In that case you're right, yeah I forgot we do that :facepalm: